codegenerator/cli/templates/static/codegen/src/ReorgDetection.res

```diff
@@ -178,6 +178,14 @@ module LastBlockScannedHashes: {
     ~confirmedBlockThreshold: int,
   ) => lastBlockScannedData.blockNumber < currentHeight - confirmedBlockThreshold
 
+  type rec trampoline<'a> = Data('a) | Callback(unit => trampoline<'a>)
+
+  let rec trampoline = value =>
```

Comment:
Could you leave a comment saying that the function optimised by ReScript to be a `while` loop, which helps us to prevent max call stack issue.

Took me some time to figure out why it is even an improvement 😅

Reply:
Ah sorry, I thought it would be implied in the name of the function that I was using trampolines to avoid stack frame assignment.

I'll add a note